src/installer/destinations.py

```diff
@@ -101,40 +102,33 @@ def finalize_installation(
         raise NotImplementedError
 
 
+@dataclass
 class SchemeDictionaryDestination(WheelDestination):
-    """Destination, based on a mapping of {scheme: file-system-path}."""
+    """Destination, based on a mapping of {scheme: file-system-path}.
+
+    Construct a ``SchemeDictionaryDestination`` object.
+
+    :param scheme_dict: a mapping of {scheme: file-system-path}
+    :param interpreter: the interpreter to use for generating scripts
+    :param script_kind: the "kind" of launcher script to use
+    :param hash_algorithm: the hashing algorithm to use, which is a member
+        of :any:`hashlib.algorithms_available` (ideally from
+        :any:`hashlib.algorithms_guaranteed`).
+    :param bytecode_optimization_levels: Compile cached bytecode for
+        installed .py files with these optimization levels. The bytecode
+        is specific to the minor version of Python (e.g. 3.10) used to
+        generate it.
+    :param destdir: A staging directory in which to write all files. This
```

Comment:
Ah, I meant documenting each variable rather than having it in the class docstring...

```py
@dataclass
class Foo:
    """The awesome Foo class

    This is very important.
    """

    one: str
    """The first string"""

    two: int
    """The second integer"""
```

Reply:
Ohhhhh, I see what you mean. That's pretty neat.

Quick question: for multiline docs, e.g. hash_algorithm, how would the variable docstring be formatted?

Reply:
```py
var: type
"""
one line
two line
"""
```

Reply:
Done.